third_party/terraform/resources/resource_compute_router_nat.go

```diff
@@ -0,0 +1,385 @@
+package google
+
+import (
+	"fmt"
+	"log"
+
+	"strings"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+	computeBeta "google.golang.org/api/compute/v0.beta"
+
+	"google.golang.org/api/googleapi"
+)
+
+var (
+	subnetworkConfig = &schema.Resource{
```

Comment:
nit: since variables are package-scoped, I'd rename this something like `routerNatSubnetworkConfig`